src/resources/spi_linux.cc

```diff
@@ -32,12 +33,34 @@ extern Object* return_open_error(Process* process, int err);
 
 MODULE_IMPLEMENTATION(spi_linux, MODULE_SPI_LINUX);
 
+// There is no close primitive, as the one from the file module is used.
 PRIMITIVE(open) {
-  ARGS(cstring, pathname);
+  ARGS(cstring, pathname, int, frequency, int, mode);
+  if (frequency <= 0) FAIL(INVALID_ARGUMENT);
   // We always set the close-on-exec flag otherwise we leak descriptors when we fork.
   // File descriptors that are intended for subprocesses have the flags cleared.
   int fd = open(pathname, O_CLOEXEC | O_RDWR);
   if (fd < 0) return return_open_error(process, errno);
+  int result = ioctl(fd, SPI_IOC_WR_MAX_SPEED_HZ, &frequency);
+  if (result < 0) {
+    close(fd);
+    return return_open_error(process, errno);
```

Comment:
Not sure about using "return_open_error" here.
Do we have something that gets the perror?

Reply:
`Primitive::os_error`

Reply:
done.